src/yb/yql/pgwrapper/pg_libpq-test.cc

```diff
@@ -190,6 +190,36 @@ TEST_F(PgLibPqTest, Simple) {
   ASSERT_EQ(row, (decltype(row){1, "hello"}));
 }
 
+TEST_F(PgLibPqTest, PgStatIdxScanNoIncrementOnErrorTest) {
+  auto conn = ASSERT_RESULT(Connect());
+
+  std::string createTableStmt = "create table many (";
+  for (int i = 1; i <= 30; ++i)
+    createTableStmt += Format("c$0 int,", i);
+  createTableStmt += "k int primary key)";
+
+  std::string createIndexStmt = "create index on many (";
+  for (int i = 1; i <= 30; ++i)
+    createIndexStmt += Format("c$0,", i);
+  createIndexStmt += "c1 asc)";
+
+  std::string predicateErrorQuery = "select * from many where ";
+  for (int i = 1; i <= 30; ++i)
+    predicateErrorQuery += Format("c$0 = 1 and c$0 < 2 and c$0 > 0 and ", i);
+  predicateErrorQuery += " true";
+
+  ASSERT_OK(conn.Execute(createTableStmt));
+  ASSERT_OK(conn.Execute(createIndexStmt));
+
+  ASSERT_NOK(conn.Fetch(predicateErrorQuery));
+  const auto noIncrementRows = ASSERT_RESULT((conn.FetchRows<int64_t>("select idx_scan from pg_stat_user_indexes")));
+  ASSERT_EQ(noIncrementRows, (decltype(noIncrementRows){0, 0}));
+
+  ASSERT_TRUE(ASSERT_RESULT(conn.HasIndexScan("select * from many where c1 = 1")));
```

Comment:
should demonstrate a large number of predicates + no error.  Get it as many predicates as you can before error.  Should be [the number of predicates in error case] - 1.  could share the same query (sub)string between both this non-error case and error case since they can be mostly the same.